doc/arch/006-forward-compatibility-for-enums.md

```diff
@@ -0,0 +1,55 @@
+# ADR: Forward compatibility for persistable enums from Server responses
+
+
+## Status
+
+Accepted
+
+
+## Context
+
+RESTful APIs tend to return enumerated values to represent properties with a fixed number of well-defined values. These are usually represented as `enum` classes on the client. When new business use cases are introduced, there could be additions to these values. If these values are consumed on the fly and are not persisted on the client, they could simply be treated as a special case (unknown) or ignored all together.
+
+In Simple, some of these values have to be persisted in the database. Hence, these enumerated values cannot be represented using a Kotlin `enum` class.
```

Comment:
We should probably also add a section that explains the additional "offline-first" constraint on the product that stops us from being able to clear the data and sync again when an app update happens. This means that:

- Apps on an older app version should be able to handle new `enum` values added to a type even though they don't know how to interpret the new value. This *could* be solved by adding an `UNKNOWN` fallback value which the server adds as well, except
- Records are shared between different users and thus, between different app versions. A newer app version that understands the new `enum` value could create a record with the new value and upload it, but an older app version that does not know the new value will overwrite it with `UNKNOWN` and this value will overwrite the actual value on the server as well when it syncs.
- An app on an older version could have a degraded, fallback experience for `enum` values it does not understand, but as soon as it updates, the app should be able to interpret the new values as if it was synced afresh.